content/guides/kubernetes/ytt-gs.md

```diff
@@ -0,0 +1,269 @@
+  #@ end
+#@ end
+```
+
+ytt comments are then used to calculate Starlark expressions to insert and concatenate this name, label, and version information into YAML nodes in multiple places like the metadata, labelSelector, and container image tag.
+
+```yaml
+#@ version
+#@ labels(with_version=True)
+#@ app_name + "-service"
+```
+
+Look at file and run in ytt playground:
+https://carvel.dev/ytt/#example:example-extract-yaml-fragments 
+
+Or run locally:
+```shell
+ytt -f playground/getting-started/example-extract-yaml-fragments/config.yml
+```
+
+## Load
+We saw `#@ load` when we were first looking at directives.
+So what does `load` do? 
+Basically it is used to load functions from other libraries. These could be [built-in modules](https://carvel.dev/ytt/docs/latest/lang-ref-ytt/) or ones you define in another file.
+```
+#@ load("@ytt:template", "template")
+#@ load("@ytt:data", "data")
+
+
+#@ load("helpers.lib.yml", "mod")
+
+```
+
+## Overlays
+ytt has a feature and module called [overlay](https://carvel.dev/ytt/docs/latest/lang-ref-ytt-overlay/#overview) which allows you to patch YAML structures. 
+By patching, you could use it for matching, inserting, replacing or removing items, etc. 
+They are used to update YAML nodes which don’t have any externally exposed ways to configure data.
+Overlays are applied in the order provided. 
+
+To use an overlay, you need to load the built-in overlay function.
+```shell
+#@ load("@ytt:overlay", "overlay")
+```
+
+In the `playground/basics/example-overlay-files` directory we have four files. 
+Two are clipped Kubernetes YAML files and the other two have ytt functionality. 
+We can see that both ops files are using the above load directive at the beginning. 
+Then there is an overlay action to do a match so the operations are only performed on specific data. For these it’s checking the name of the resource e.g.
+```
+#@overlay/match by=overlay.subset({"metadata":{"name":"example-ingress1"}})
+```
+
+The other parts either are removing items or adding lines if they’re missing.
+```yaml
+#@overlay/remove
+ingress.kubernetes.io/rewrite-target:
+
+#@overlay/match missing_ok=True
+nginx.ingress.kubernetes.io/enable-access-log: "true"
+```
+
+Look at files and run in interactive playground:
+https://carvel.dev/ytt/#example:example-overlay-files 
+
+Or run locally:
+```shell
+ytt -f playground/basics/example-overlay-files
+```
+
+## Data Values and Schema
+The data module, `#@ load("@ytt:data", "data")`, can be used to give users a configuration interface. It allows you to pass in files with a schema module, `#@data/values-schema`, or via command line. To see all of the options run `ytt -h` to see flags.
+
+Note: If you are using an old version of ytt, the following ytt commands will fail.
+
+The k8s-add-global-label/add-label.yml file uses the annotation `​​#@ data.values.build_num` along with an overlay to check if it’s missing. If it’s missing it will be filled in with the line below.
```

Comment:
"k8s-add-global-label/add-label.yml" should probably be formatted as in-line code (`k8s-add-global-label/add-label.yml`)